test/jit/test_enum.py

```diff
@@ -262,17 +267,60 @@ def forward(self):
 
         self.assertEqual(scripted(), Color.RED.value)
 
+    def test_enum_return(self):
+        global Color
+
+        class Color(Enum):
+            RED = 1
+            GREEN = 2
+
+        def return_enum(cond: bool):
+            if cond:
+                return Color.RED
+            else:
+                return Color.GREEN
+
+        # TODO(gmagogsfm): Re-enable hooks when serialization/deserialization
+        # is supported.
+        with torch._jit_internal._disable_emit_hooks():
+            scripted = torch.jit.script(return_enum)
+
+        self.assertEqual(scripted(True), Color.RED)
+        self.assertEqual(scripted(False), Color.GREEN)
+
+    def test_enum_module_return(self):
+        global Color
+
+        class Color(Enum):
+            RED = 1
+            GREEN = 2
+
+        class TestModule(torch.nn.Module):
+            def __init__(self, e: Color):
+                super(TestModule, self).__init__()
+                self.e = e
+
+            def forward(self):
+                return self.e
+
+        m = TestModule(Color.RED)
+        with torch._jit_internal._disable_emit_hooks():
+            scripted = torch.jit.script(m)
+
+        self.assertEqual(scripted(), Color.RED)
```

Comment:
Consider running this test with all enum values to make sure the `IValue` -> `PyObject` conversion code isn't returning the first value of the enum always.